File: Claude/Simples/countingsort/countingsort.py
# Input: How to implement the countingsort sort algorithm? in python

def counting_sort(arr):
    """
    Perform counting sort on the given array.
    
    Args:
        arr (list): The input array to be sorted.
        
    Returns:
        list: The sorted array.
    """
    # Find the maximum element in the array
    max_element = max(arr)
    
    # Create a count array to store the count of individual
    # elements in the input array
    count_arr = [0] * (max_element + 1)
    
    # Store count of each element
    for element in arr:
        count_arr[element] += 1
    
    # Modify the count_arr to get the actual
    # position of this element in the output array
    for i in range(1, len(count_arr)):
        count_arr[i] += count_arr[i - 1]
    
    # Build the output character array
    output_arr = [0] * len(arr)
    i = len(arr) - 1
    while i >= 0:
        output_arr[count_arr[arr[i]] - 1] = arr[i]
        count_arr[arr[i]] -= 1
        i -= 1
    
    return output_arr

lista = [1, 5, 2, 25, 40, 10, 0]

counting_sort(lista)
print(lista)

"""
Score: 1
"""

File: Claude/Simples/countingsort/countingsort.cpp
// Input: How to implement the countingsort sort algorithm? in c++

#include <iostream>
#include <vector>
#include <algorithm>

void countingSort(std::vector<int>& arr) {
    if (arr.empty()) {
        return;
    }

    // Find the range of the array
    int max = *std::max_element(arr.begin(), arr.end());
    int min = *std::min_element(arr.begin(), arr.end());
    int range = max - min + 1;

    // Create a count array to store count of individual numbers
    std::vector<int> count(range);

    // Create an output array
    std::vector<int> output(arr.size());

    // Store count of each number
    for (int i = 0; i < arr.size(); i++) {
        count[arr[i] - min]++;
    }

    // Change count[i] so that count[i] now contains actual
    // position of this number in output array
    for (int i = 1; i < count.size(); i++) {
        count[i] += count[i - 1];
    }

    // Build the output array
    for (int i = arr.size() - 1; i >= 0; i--) {
        output[count[arr[i] - min] - 1] = arr[i];
        count[arr[i] - min]--;
    }

    // Copy the output array to arr, so that arr now
    // contains sorted numbers
    for (int i = 0; i < arr.size(); i++) {
        arr[i] = output[i];
    }
}

// Driver program to test above function
int main() {
    std::vector<int> arr = {64, 34, 25, 12, 22, 11, 90};
    
    std::cout << "Original array: ";
    for (int num : arr) {
        std::cout << num << " ";
    }
    std::cout << std::endl;

    countingSort(arr);

    std::cout << "Sorted array: ";
    for (int num : arr) {
        std::cout << num << " ";
    }
    std::cout << std::endl;

    return 0;
}

/*
Score: 1
*/